_posts/2023-05-25-Announcing-Data-Prepper-2.3.0.md

```diff
@@ -0,0 +1,147 @@
+
+Data Prepper 2.3 supports three types of expressions
+
+* Conditional expressions
+    * Conditional expressions evaluate to a result of Boolean type. The expressions can now have functions in them. For example, `length(/message) > 20` would evaluate to true if length of the message field in the event is greater than 20 otherwise it evaluates to false. 
+* Arithmetic expressions
+    * Arithmetic expressions evaluate to a result of Integer or Float type. The expressions can have simple arithmetic operators like `+,-,*,/` with functions or JSON pointers or literals as operands. For example, the following expression will add the length of message field of type string in the event with the value of event metadata with key `integerField` and subtracts 1 from it
+
+ ```length(/message) + getMetadata("integerField") - 1```
+
+* String expressions
+    * String expressions evaluate to a result of String type. String concatenation operator is supported in addition to using functions or JSON pointers or literals as operands. For example, the following expression will add the message1 field of type string in the event with message2 field of type string in the event and appends suffix to it.
+
+```/message1 + /message2 + "suffix"```
+
+## Event tagging
+
+Data Prepper 2.3 supports tagging events while using grok processor. Events can be tagged optionally using the following configuration
+
+```
+processor:
+   - grok: 
+        match:
+          message: <pattern>
+        tags_on_match_failure: ["grok_match_fail", "another_tag"]
+```
+
+Presence of tags can be checked in conditional expressions in different processors or routing using `hasTags()` function to do conditional ingestion. For example, a conditional expression checking for presence of tag `grok_match_fail` would be `hasTags("grok_match_fail")`
+
+## Enhancements to `add_entries` processor
+
+### Expression based value
+
+`add_entries` processor is enhanced to support adding values based on expression where the return type can be Boolean or Integer/Float or String. An example `add_entries` processor configuration with `value_expression` option
+
+```
+processor:
+   - add_entries:
+       entries:
+         - key: "request_len"
+           value_expression: "length(/request)"
+```
+
+This configuration adds an entry with key `request_len` with value equal to the length of the request field in the key. The value expression can be any of the expressions supported (see [expression syntax](https://github.com/opensearch-project/data-prepper/blob/main/docs/expression_syntax.md) for more details)
+
+### Setting event metadata keys
+
+`add_entries` processor can also add entries in the event’s metadata instead of the event itself. And example `add_entries` processor configuration for adding an entry to metadata 
```

Comment:
```suggestion
The `add_entries` processor can add entries in the event’s metadata instead of the event itself. The following example shows an `add_entries` processor configuration for adding an entry to metadata: 
```